subsys/bootloader/include/fprotect.h

```diff
@@ -10,7 +10,18 @@
 #include <stdint.h>
 #include <string.h>
 
-/* Flash protect specified area */
-void fprotect_area(u32_t start, size_t length);
+/**
+ * @brief Protect flash area against writes.
+ *
+ * @param[in]  start   Start of range to protect.
+ * @param[in]  length  Length in bytes of range to protect.
+ *
+ * @retval 0        On success.
+ * @retval -EFAULT  If any of the argument are incorrect.
+ * @retval -EAGAIN  If function is called too many times. Applies to
```

Comment:
`EAGAIN` signals that the operation can be retried. If this is not the case here, consider using another error.

Reply:
Which do you prefer? Based on the doc in `zephyr/lib/libc/minimal/include/errno.h` it is not easy to find an exact match.

`ENXIO`
`ENOMEM`
`ENODEV`

Reply:
in zephyr errno i see `#define EAGAIN 11       /* No more contexts */`, so looks like this fit here. Also -ENOSPC seams to be suitable here.

Reply:
@nvlsianpu I disagree, `EAGAIN `means: try again, it might succeed at a later point; generally this happens when a resource is temporarily unavailable. Here, we have the exact opposite, this function will _surely_ fail if called again.